File: 125-ValidPalindrome/125-ValidPalindrome.java
// Last updated: 10/12/2025, 12:23:50 AM
class Solution {
    public boolean isPalindrome(String s) {
         StringBuilder cleaned = new StringBuilder();

        for (char c : s.toCharArray()) {
            if (Character.isLetterOrDigit(c)) {       // keep only letters and digits
                cleaned.append(Character.toLowerCase(c)); // convert to lowercase
            }
        }

        int i=0;
        int j=cleaned .length()-1;

        while(i<=j){
            if(cleaned.charAt(i)==cleaned.charAt(j)){
                i++;
                j--;
            }
            else{
                return false;
            }
        }
        return true;
    }
}

File: 125-ValidPalindrome/125-ValidPalindrome.cpp
// Last updated: 10/12/2025, 12:21:27 AM
class Solution {
public:
    bool isPalindrome(string s) {
        string temp;
        for(char c:s){
            if(isalnum(c)){
                temp += tolower(c);
            }
        }

        int i=0;
        int j=temp.length()-1;

        while(i<=j){
            if(temp[i]==temp[j]){
                i++;
                j--;
            }
            else{
                return false;
            }
        }
        return true;
    }
};